torch/csrc/jit/passes/op_replacement.cpp

```diff
@@ -0,0 +1,65 @@
+#include <c10/util/Exception.h>
+#include <torch/csrc/jit/ir/irparser.h>
+#include <torch/csrc/jit/operator_upgraders/upgraders.h>
+#include <torch/csrc/jit/operator_upgraders/utils.h>
+#include <torch/csrc/jit/operator_upgraders/version_map.h>
+#include <torch/csrc/jit/passes/op_replacement.h>
+#include <torch/csrc/jit/runtime/graph_iterator.h>
+#include <limits>
+#include <regex>
+#include <string>
+
+namespace torch {
+namespace jit {
+
+struct OpsReplacer {
+  OpsReplacer(std::shared_ptr<Graph> graph) : graph_(std::move(graph)) {}
+
+  void run() {
+    if (!graph_->get_op_version().has_value()) {
+      return;
+    }
+    auto current_version = graph_->get_op_version().value();
+    DepthFirstGraphNodeIterator graph_it(graph_);
+    Node* node = graph_it.next();
+    while (node) {
+      if (auto schema = node->maybeSchema()) {
+        auto schema_name = schema->name() + "." + schema->overload_name();
+        // this implies there was a version bump because of this operator
+        auto version_entry = operator_version_map.find(schema_name);
+        if (version_entry != operator_version_map.end()) {
+          auto upgrader_name =
+              findUpgrader(version_entry->second, current_version);
+          auto upgrader_graph_entry =
+              upgraders_graph.find(upgrader_name.upgrader_name);
+          assert(upgrader_graph_entry != upgraders_graph.end());
```

Comment:
do we need to always throw when upgrader is not available? (otherwise assert compiled out in opt mode)